### ISSUE TITLE
BUG: Switching layers using a streamdeck button causes error

### ISSUE THREAD

**Issue by matt-cahill** [DCS-Skunkworks/DCSFlightpanels#64]
```
02.05.2020 10:14:26  version : 1.0.218.6063

System.InvalidOperationException
The calling thread cannot access this object because a different thread owns it.
   at System.Windows.Threading.Dispatcher.VerifyAccess()
   at DCSFlightpanels.PanelUserControls.UserControlStreamDeckUIBase.HideAllDotImages()
   at DCSFlightpanels.PanelUserControls.UserControlStreamDeckUIBase.UpdateButtonInfoFromSource()
   at DCSFlightpanels.PanelUserControls.UserControlStreamDeckUIBase.UIShowLayer(String layerName)
```
![](https://media.discordapp.net/attachments/700674160304848906/705912325890375720/unknown.png)

**Comment by matt-cahill** [comment]
confirmed fixed in 4.0.30

**Comment by matt-cahill** [comment]
sorry thought this was fixed but still getting this on a fresh profile

```
03.05.2020 12:21:53  version : 1.0.226.7172

System.InvalidOperationException
The calling thread cannot access this object because a different thread owns it.
   at System.Windows.Threading.Dispatcher.VerifyAccess()
   at System.Windows.DependencyObject.SetValue(DependencyProperty dp, Object value)
   at DCSFlightpanels.MainWindow.SetWindowState()
   at DCSFlightpanels.MainWindow.PanelSettingsChanged(Object sender, PanelEventArgs e)
```

**Comment by matt-cahill** [comment]
In 4.0.39 this error is gone however switching layers from a streamdeck button now triggers the "something changed, this profile needs to be saved" state in the UI.

If subsequently saved the only thing that changes in the file is the time stamp

```diff
#This file can be manually edited using any ASCII editor.
-#File created on 4/05/2020 12:00:00 AM 4/05/2020 6:19:51 PM#
+#File created on 4/05/2020 12:00:00 AM 4/05/2020 6:23:53 PM#
#
#IMPORTANT INFO REGARDING the keyboard key AltGr (RAlt as named in DCS) or RMENU as named DCSFP
#When you press AltGr DCSFP will register RMENU + LCONTROL. This is a bug which "just is". You need to modify that in the profile
#by deleting the + LCONTROL part.
#So for example AltGr + HOME pressed on the keyboard becomes RMENU + LCONTROL + HOME
#Open text editor and delete the LCONTROL ==> RMENU + HOME
```

**Comment by matt-cahill** [comment]
fixed in 4.0.52
